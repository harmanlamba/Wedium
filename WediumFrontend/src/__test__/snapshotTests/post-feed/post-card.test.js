import React from 'react';
import { shallow } from 'enzyme';

import PostCard from '../../../components/post-feed/post-card';

it('snapshot PostCard default load', () => {
  const posts = [
    {
      articleImageUrl: 'Image URL',
      articleUrl: 'Article URL',
      date: '',
      description: 'Post Description',
      isFavourited: false,
      isPostLiked: false,
      numberOfLikes: 2,
      postId: 3028,
      postType: 'Culture',
      title: 'Post Title',
      username: 'Bob Bobby',
    },
  ];

  const setRouteLeaveHook = jest.fn();

  let wrapper = shallow(
    <PostCard.WrappedComponent
      post={posts[0]}
      params={{
        router: setRouteLeaveHook,
      }}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
