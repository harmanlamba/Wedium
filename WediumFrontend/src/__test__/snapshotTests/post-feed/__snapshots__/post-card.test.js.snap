// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot PostCard default load 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
  onClick={[Function]}
  onMouseDown={[Function]}
>
  <div>
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-cardContent-2"
    >
      <WithStyles(ForwardRef(Typography))
        variant="subtitle2"
      >
        <span
          className="makeStyles-postType-3"
        >
          Culture
        </span>
        <WithStyles(ForwardRef(Link))
          className="makeStyles-articleLink-7"
          href="Article URL"
          onClick={[Function]}
          target="_blank"
        >
          undefined
        </WithStyles(ForwardRef(Link))>
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-postTitle-4"
        variant="h6"
      >
        Post Title
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Link))
        color="textSecondary"
        variant="subtitle2"
      >
        Bob Bobby
      </WithStyles(ForwardRef(Link))>
       
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        variant="subtitle2"
      >
        Invalid date
         - 
        Invalid date
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </div>
  <div
    className="makeStyles-rightPanel-8"
  >
    <div
      className="makeStyles-buttonGroup-9"
      onClick={[Function]}
    >
      <WithStyles(Connect(PostLike))
        isPostLiked={false}
        numberOfLikes={2}
        postId={3028}
      />
      <WithStyles(Connect(withRouter(PostFavourite)))
        isFavourited={false}
        postId={3028}
      />
    </div>
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-image-6"
      component="img"
      src="Image URL"
    />
  </div>
</WithStyles(ForwardRef(Card))>
`;
